src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -52,140 +53,113 @@ public DatabasesLandlord(ServerStore serverStore)
+                            UnloadDatabase(databaseName, dbRecordIsNull: true);
+                            return;
+                        }
 
-                if (DatabasesCache.TryGetValue(databaseName, out var task) == false)
-                {
-                    // if the database isn't loaded, but it is relevant for this node, we need to create
-                    // it. This is important so things like replication will start pumping, and that 
-                    // configuration changes such as running periodic backup will get a chance to run, which
-                    // they wouldn't unless the database is loaded / will have a request on it.          
-                    task = TryGetOrCreateResourceStore(databaseName);
-                }
+                        if (ShouldDeleteDatabase(context, databaseName, rawRecord))
+                            return;
 
-                if (task.IsCanceled || task.IsFaulted)
-                    return;
+                        topology = rawRecord.GetTopology();
+                        if (topology.RelevantFor(_serverStore.NodeTag) == false)
+                            return;
 
-                switch (changeType)
-                {
-                    case ClusterDatabaseChangeType.RecordChanged:
-                        if (task.IsCompleted)
+                        if (rawRecord.IsDisabled() || rawRecord.GetDatabaseStateStatus() == DatabaseStateStatus.RestoreInProgress)
                         {
-                            NotifyDatabaseAboutStateChange(databaseName, task, index);
-                            if (type == ClusterStateMachine.SnapshotInstalled)
-                            {
-                                NotifyPendingClusterTransaction(databaseName, task, index, changeType);
-                            }
-
+                            UnloadDatabase(databaseName);
                             return;
                         }
+                    }
 
-                        task.ContinueWith(done =>
-                        {
-                            NotifyDatabaseAboutStateChange(databaseName, done, index);
+                    if (DatabasesCache.TryGetValue(databaseName, out var task) == false)
+                    {
+                        // if the database isn't loaded, but it is relevant for this node, we need to create
+                        // it. This is important so things like replication will start pumping, and that 
+                        // configuration changes such as running periodic backup will get a chance to run, which
+                        // they wouldn't unless the database is loaded / will have a request on it.          
+                        task = TryGetOrCreateResourceStore(databaseName);
+                    }
+
+                    if (task.IsCanceled || task.IsFaulted)
+                        return;
+
+                    var database = await task;
```

Comment:
If this fails, the user is going to get an error, but does it impact anything else?

Reply:
It might only impact on the server start-up, I guess it is better to create a task for each database.